satpy/readers/hdfeos_base.py

```diff
@@ -231,9 +231,9 @@ def _scale_and_mask_data_array(self, data, is_category=False):
         # don't scale category products, even though scale_factor may equal 1
         # we still need to convert integers to floats
         if scale_factor is not None and not is_category:
-            data = data * np.float32(scale_factor)
             if add_offset is not None and add_offset != 0:
-                data = data + add_offset
+                data = data - add_offset
+            data = data * np.float32(scale_factor)
```

Comment:
As you noted in the PR description, this is quite unexpected, so how about putting a comment here explaining what is happening, maybe with a link to the reference?

Reply:
I added a docstring with a link to the L1b PDF. Even in that document it is hard to find the information. It is also mentioned in L2 documentation more clearly but I thought L1b was a better source of truth to list.